server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -145,9 +157,12 @@ public void test_jwt_authentication() throws Exception {
         assertThat(credentials.username()).isNull();
         credentials.setUsername(JWT_USER.name());
 
-        Role authenticatedRole = jwtAuth.authenticate(credentials, null);
+        Role authenticatedRole = jwtAuth.authenticate(credentials, CONNECTION_PROPERTIES);
         assertThat(authenticatedRole).isNotNull();
         assertThat(authenticatedRole.name()).isEqualTo(JWT_USER.name());
+        // Assure sure JwkProvider for the given iss is cached
```

Comment:
I think we can extend this test and imitate real scenario

Imitate new connection - create new `JWTAuthenticationMethod` instance. 
Second instance can use `() -> null` instead of `jwkProviderFunction(null)` and we can say in a code comment, that without cache if throws NPE.
Then pass same cache (with 1 entry fro the prev call) and 
do same assertion that user is authenticated with the second  JWTAuthenticationMethod instance and no NPE is thrown.

Reply:
👍